### PR TITLE
feat: add batch actor upgrade

### PR DESCRIPTION
**Pull request by MasterPtato**
Fixes RVT-4155

### CODE REVIEW

**Comment by MasterPtato** [comment]
* **#1480** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/1480?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/1480?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: packages/services/ds/src/ops/server/list_for_env.rs [hunk 1]
```diff
@@ -8,6 +8,7 @@ pub struct Input {
 	pub tags: HashMap<String, String>,
 	pub include_destroyed: bool,
 	pub cursor: Option<Uuid>,
+	pub limit: usize,
```
Comment (NathanFlurry) [inline packages/services/ds/src/ops/server/list_for_env.rs:11]: these limits need some sort of cap on it

File: packages/api/actor/src/route/actors.rs [hunk 2]
```diff
@@ -439,6 +440,98 @@ pub async fn upgrade(
...
+		);
+	}
+
+	let build_id = resolve_build_id(&ctx, env_id, body.build, body.build_tags.flatten()).await?;
+
+	// Work in batches
+	let mut count = 0;
+	let mut cursor = None;
+	loop {
+		let list_res = ctx
```
Comment (NathanFlurry) [inline packages/api/actor/src/route/actors.rs:488]: this is going to be problematic, it's very likely that applications will have > 10,000 actors. we need to stream the actors with a cursor. i'm fine with adding a ticket to do this post-mvp as long as it has a due date.

**Review by NathanFlurry — CHANGES_REQUESTED** [review]

File: packages/services/ds/src/ops/server/list_for_env.rs (hunk 1 above)
Comment (MasterPtato) [reply packages/services/ds/src/ops/server/list_for_env.rs:11]: This is an internal op, the limits should go on the api layer

File: packages/api/actor/src/route/actors.rs (hunk 2 above)
Comment (MasterPtato) [reply packages/api/actor/src/route/actors.rs:488]: This is already chunking the actors with a cursor, its just doing so in chunks of 10k

Comment (NathanFlurry) [reply packages/api/actor/src/route/actors.rs:488]: i'm blind

**Review by NathanFlurry — APPROVED** [review]

**Pull request closed without merging**